### PR TITLE
[airlift] Initial README update pass

### PR DESCRIPTION
**Pull request by benpankow**
## Summary & Motivation

Updates README instructions, code samples and adds some screenshots.

## Changelog [Docs]

`NOCHANGELOG`

### CODE REVIEW

**Comment by benpankow** [comment]
* **#23974** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23974?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/README.md [hunk 1]
```diff
@@ -36,45 +36,49 @@ Airlift depends on the the availability of Airflow’s REST API. Airflow’s RES
...
 ```bash
 pip install uv
-uv pip install dagster-airlift[core] dagster-webserver
+uv pip install 'dagster-airlift[core]' dagster-webserver
 ```
 
-Next, you should create a `Definitions` object using `build_defs_from_airflow_instance`.
+Next, create a new Python file to hold your Dagster code. Create a `Definitions` object using `build_defs_from_airflow_instance`:
 
+`airlift.py`
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/README.md:80]: The rendering of this standalone is a little goofy:

File: examples/experimental/dagster-airlift/README.md [hunk 2]
```diff
@@ -145,79 +169,102 @@ defs = Definitions.merge(
...
-The `from_task` and `from_dag` utilities link created assets to their underlying tasks. Assets which are properly linked will be materialized by the Airlift sensor once the corresponding task completes.
 
 ### Viewing observed assets
 
 Once your assets are set up, you should be able to reload your Dagster definitions and see a full representation of the dbt project and other data assets in your code.
 
-Kicking off a run of the DAG, you should see the newly created assets materialize in Dagster.
+<p align="center">
+
+![Observed asset graph in Dagster](./images/observe.svg)
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/README.md:262]: how are you making these svgs these are great

File: examples/experimental/dagster-airlift/README.md [hunk 3]
```diff
@@ -273,31 +327,30 @@ The migration file acts as the source of truth for migration status. A task whic
...
+<p align="center">
+
+![dbt build executing in Dagster](./images/migrated_dag.png)
+
+</p>
+
 #### Migrating custom operators
 
-For all other operator types, we recommend creating a new factory class whose arguments match the inputs to your Airflow operator. Then, you can use this factory to build definitions for each Airflow task.
+For all other operator types, we recommend creating a new factory class whose arguments match the inputs to your Airflow operator. Then, you can use this factory to build asset definitions for each Airflow task.
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/README.md:353]: We no longer have this fully supported factory class thing, so we can just say a factory function.

Comment (schrockn) [reply examples/experimental/dagster-airlift/README.md:353]: Also we can just say "build definitions" for each Airflow task

**Review by schrockn — APPROVED** [review]
I have a couple comments from a quick readthrough but good improvment. Make sure to run `make prettier` prior to merge!

File: examples/experimental/dagster-airlift/README.md (hunk 2 above)
Comment (benpankow) [reply examples/experimental/dagster-airlift/README.md:262]: Bottom button in the asset graph! 
Screenshot 2024-08-27 at 1 20 24 PM
great feature

Comment (schrockn) [reply examples/experimental/dagster-airlift/README.md:258]: TIL!

**Pull request merged**